degica/kaiser
# Fix http wait issues

## Context

Often times environments marked `http` will wait forever because a database migration is not run, or the database cannot be connected to. This PR solves that problem by first checking if the server is reachable, and then checking to see if the server returns a 200 OK. If the thing is 200 OK Kaiser should indicate it starts correctly, otherwise, terminate the wait script and throw an error.

## How to test
Easiest way to test this is with hats because its got the http mark and had this issue
1. `rake build` in this repo and `gem install kaiser/pkg/kaiser-0.4.4.gem`
2. Go to hats and `kaiser up`
3. Observe that it should say `Started Successfully`
4. Go to the hats repo an `kaiser login sh` and run `rails db:drop`
5. Now run `kaiser up` again
6. It should instead of waiting infinitely, now say `Failed with HTTP status: 500`
